stratify/_vinterp.pyx

```diff
@@ -544,10 +543,6 @@ cdef class _Interpolation(object):
         fz_src = fz_src.astype(np.float64)
 
         # Broadcast the z_target shape if it is 1d (which it is in most cases)
-        if z_target.ndim == 1:
-            z_target_size = z_target.shape[0]
-        else:
-            z_target_size = z_target.shape[axis]
```

Comment:
We don't want to index `z_target` here with `axis`, as `axis` is relative to `fz_src`.

We want to use `zp_axis` defined later on [here](https://github.com/SciTools-incubator/python-stratify/pull/13/files#diff-8784b747a5ef9177cf627ad895736105R549), as that is the `axis` relative to `z_src` - and for the `nd` case we are going to ensure that the dimensionality of `z_target` and `z_src` are in sync, see [here](https://github.com/SciTools-incubator/python-stratify/pull/13/files#diff-8784b747a5ef9177cf627ad895736105R561) onwards